pkg/reconciler/instances/istio/actions/performer.go

```diff
@@ -527,3 +539,52 @@ func IsSidecarInjectionNamespacesByDefaultEnabled(workspace chart.Factory, branc
 
 	return enableNamespacesByDefault, nil
 }
+
+func isCNIEnabled(workspace chart.Factory, branch string, istioChart string) (bool, error) {
+	ws, err := workspace.Get(branch)
+	if err != nil {
+		return false, err
+	}
+
+	istioHelmChart, err := loader.Load(filepath.Join(ws.ResourceDir, istioChart))
+	if err != nil {
+		return false, err
+	}
+
+	mapAsJSON, err := json.Marshal(istioHelmChart.Values)
+	if err != nil {
+		return false, err
+	}
+	var chartValues chartValues
+
+	err = json.Unmarshal(mapAsJSON, &chartValues)
+	if err != nil {
+		return false, err
+	}
+
+	return chartValues.Components.CNI.Enabled, nil
+}
+
+func isCNIRolloutRequired(context context.Context, kubeClient k8sClient.Interface, workspace chart.Factory, branchVersion string, istioChart string, logger *zap.SugaredLogger) (bool, error) {
```

Comment:
We need to consider what we pass in the helm chart for the desired state and what we have in CM override, the third state would be the current state on the cluster, but currently, we don't support it in the reconciler, so we focus on applying CNI config right now.

Reply:
Also, this is a valid point, but because of how this Istio reconciler is split we can't make passing logic to the post action easily.
I would love to actually create a pre-action for all of those configuration merges to istio operator and then invoking actions accordingly, but this needs to be done in later steps.